jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java

```diff
@@ -1884,7 +1884,7 @@ else if (arg.toLowerCase(Locale.ENGLISH).endsWith(".properties"))
             List<Object> objects = new ArrayList<>(args.length);
             for (String arg : args)
             {
-                if (!arg.toLowerCase(Locale.ENGLISH).endsWith(".properties") && (arg.indexOf('=') < 0))
+                if (arg.toLowerCase(Locale.ENGLISH).endsWith(".xml"))
```

Comment:
Why is not this loop merged with the previous?

Also, the logic about `indexOf('=')` would be broken in case of system properties with no values (it would be broken in 12.0.x too)?

Reply:
it could be, I'm just going for minimal changes in Jetty 10.x

Regarding Jetty 12, this isn't an issue there.
The entire processing of XmlConfiguration arguments was rewritten with environments in mind.
The arg processing in Jetty 12 is far more sane.

Reply:
Can we have an else warning?